images/helm-operator/run-operator.sh

```diff
@@ -259,38 +261,45 @@ while true; do
             HELM_ARGS+=("-f" "$EXTRA_VALUES_FILE")
         fi
 
-        if [ -z "$RESOURCE_VALUES" ]; then
-            echo "No values for $FULL_RESOURCE_NAME, using default values"
-        else
-            VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values.yaml"
-            echo -E "$RESOURCE_VALUES" > "$VALUES_FILE"
+        VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values.yaml"
+        NEW_VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values-NEW.yaml"
 
-            HELM_ARGS+=("-f" "$VALUES_FILE")
-        fi
+        echo -E "$RESOURCE_VALUES" > "$NEW_VALUES_FILE"
 
-        mkdir -p "$RESOURCE_VERSIONS_DIRECTORY"
-        RESOURCE_VERSION_FILE="$RESOURCE_VERSIONS_DIRECTORY/$RESOURCE_NAMESPACE-$RESOURCE_NAME.resourceVersion"
-        # If the resource version for this Release CR hasn't changed, we can skip running helm upgrade.
-        if [[ -s "$RESOURCE_VERSION_FILE" && "$(cat "$RESOURCE_VERSION_FILE")" == "$RESOURCE_RESOURCE_VERSION" ]]; then
+        # If the spec for this CR hasn't changed, we can skip running helm upgrade.
+        if [[ -a "$VALUES_FILE" && "$(cat "$VALUES_FILE")" == "$(cat "$NEW_VALUES_FILE")" ]]; then
```

Comment:
We now compare the spec of the last one we applied to the current spec, rather than comparing just resourceVersion.